### PR TITLE
Add NUOPC cap and rearrange driver directory

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Add NUOPC cap and rearrange the driver directory
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    Tested on cheyenne, full test suite, bit-for-bit 
     Also tested on a CESM2 T62_g37 DTEST compset, successfully ran 5 days with CMEPS
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:

Adds driver/nuopc/cmeps which was successfully tested with CESM2 at T62_g37 with DTEST.
Renames driver/cesm to driver/mct/cesm1
Renames driver/cice to driver/standalone/cice
Renames driver/hadgem3 to driver/subroutine/hadgem3

Refactored some aspects of the nu_diag initialization because CMEPS + NUOPC introduces the requirement that the coupling cap initializes the CICE log file.  I have provided feedback that this is not an ideal approach because it will create differences with other caps and other applications, but that largely went in one ear and out the other.  There are other ways this could be handled in CICE6, but in the end, we want to keep the initialization in the cice_init routine, so we have to be able to "detect" when nu_diag has been set earlier in those cases.

Documentation was added to provide some information about the drivers directory.

### CODE REVIEW

File: doc/source/user_guide/ug_running.rst [hunk 1]
```diff
@@ -350,9 +350,19 @@ be modified.
...
 
+There are four basis issues that need to be addressed when porting and these are addressed in four separate files in the script system,
+
+- setup of the environment such as compilers, environment variables, and other support software (in **env.[machine]_[environment]**)
+
+- setup of the Macros file to support the model build (in **Macros.[machine]_[environment]**)
+
+- setup of the batch scripts (in **cice.batch.csh**)
+
+- setup of the model launch (in **cice.launch.csh**)
```
Comment (phil-blain) [inline doc/source/user_guide/ug_running.rst]: Thanks for that! I had noticed a few months ago that we did not mention `cice.launch.csh` and it was on my backlog to do a PR for that before the tutorial!

File: doc/source/developer_guide/dg_driver.rst [hunk 2]
```diff
@@ -7,22 +7,37 @@ Driver and Coupling
...
 
-Other **cicecore/drivers/** directories are similarly implemented with a top level coupling layer,
-that is largely specified by an external coupled system and then some version of the **CICE_InitMod.F90**,
-**CICE_RunMod.F90**, and **CICE_FinalMod.F90** files.
+The files provide the top level sequencing for calling the standalone CICE model.
+
+Adding a New Driver
+------------------------
+
+The drivers directory contains two-levels of subdirectories.  The first layer indicates the coupling infrastructure or strategy and the second later indicates the application or coupler the driver is written for.  At the present time, the directory structures looks like::
```
Comment (phil-blain) [inline doc/source/developer_guide/dg_driver.rst]: "two-levels" -> "two levels"

Comment (apcraig) [reply doc/source/developer_guide/dg_driver.rst]: Good catch, this is clearly incorrect.  I have just pushed a fix.

**Comment by apcraig** [comment]
Thanks @eclare108213, I have updated the documentation again.  Will wait for checks to complete then will probably merge.

**Comment by eclare108213** [comment]
Aren't all of the driver interfaces essentially calling subroutines?

**Comment by apcraig** [comment]
@eclare108213, that's largely true unless you are running multiple executable and coupling thru a low level interface.  The difference between subroutine and mct/nuopc/esmf/oasis is the subroutine interfaces are relatively arbitrary and defined by the application, not by a formal coupling interface.  Think of an ocean model calling cice from the middle of it's subroutines with some arbitrary calls, maybe directly into cice or maybe into some layer on cice.  It's not really a formal layer of infrastructure.

I thought about whether subroutine was a good directory name and decided it was ok.  In hindsight, maybe it isn't.  We can rename the "subroutine" driver directory if you'd like.  How about "subprogram" or "other" or something else?  Anyone have any other ideas?

**Comment by eclare108213** [comment]
I was thinking the same thing.  The main idea is that this form of coupling treats the ice model much less independently, more integrated.  What about something like 'subsidiary' or 'direct' or something like that?

**Comment by apcraig** [comment]
I like direct over subroutines and have just changed that.  Good idea.

**Comment by apcraig** [comment]
Just FYI that I am going to merge this today unless there are any other comments.  I'd like to get it in before the automated testing for the weekend if it's ready.

File: configuration/scripts/cice.settings [hunk 3]
```diff
@@ -12,7 +12,7 @@ setenv ICE_OBJDIR     ${ICE_RUNDIR}/compile
 setenv ICE_RSTDIR     ${ICE_RUNDIR}/restart
 setenv ICE_HSTDIR     ${ICE_RUNDIR}/history
 setenv ICE_LOGDIR     ${ICE_CASEDIR}/logs
-setenv ICE_DRVOPT     cice
+setenv ICE_DRVOPT     standalone/cice
```
Comment (phil-blain) [inline configuration/scripts/cice.settings:15]: @apcraig I noticed that this change from `drivers/cice` to `drivers/standalone/cice` was not reflected in the cice.settings documentation, that still references `cice`. Minor, but still thought I'd mention it.

Comment (apcraig) [reply configuration/scripts/cice.settings:15]: Good catch, I'll update it in a sandbox that I plan to PR soon.  Thanks!

**Pull request merged**